packages/cli/subtasks/pick-contract.js

```diff
@@ -0,0 +1,43 @@
+const fs = require('fs');
+const { subtask } = require('hardhat/config');
+const inquirer = require('inquirer');
+const autocomplete = require('inquirer-list-search-prompt');
+const logger = require('@synthetixio/core-js/utils/logger');
+const { SUBTASK_PICK_CONTRACT } = require('../task-names');
+
+subtask(SUBTASK_PICK_CONTRACT, 'Pick contract to interact with').setAction(
+  async (taskArguments, hre) => {
+    logger.info('Pick a contract:');
+
+    const deploymentData = JSON.parse(fs.readFileSync(hre.deployer.paths.deployment));
+    const contracts = Object.keys(deploymentData.contracts);
+
+    _prioritizeTarget(contracts, 'Synthetix');
+
+    await _prompt(contracts);
+  }
+);
+
+function _prioritizeTarget(contracts, itemName) {
+  contracts.splice(contracts.indexOf(itemName), 1);
+  contracts.unshift(itemName);
+}
+
+async function _prompt(contracts) {
+  // Set up inquirer
```

Comment:
Maybe make this comment more informative? I.e. "Set up inquirer with plugin to filter results as you type (autocomplete)"